### ISSUE TITLE
RxSO3Base operator= should use quaternion_nonconst() instead of quaternion()

### ISSUE THREAD

**Issue by nburrus** [strasdat/Sophus#132]
Below is the patch I applied (sorry for inlining it since it's just one line and I have not forked the project). In short, we can't assign using `quaternion()` since it's const. I've just changed it to `quaternion_nonconst()`.

```
diff --git a/sophus/rxso3.hpp b/sophus/rxso3.hpp
index 463d1ae..d94bfc8 100644
--- a/sophus/rxso3.hpp
+++ b/sophus/rxso3.hpp
@@ -197,7 +197,7 @@ class RxSO3Base {
   template <class OtherDerived>
   SOPHUS_FUNC RxSO3Base<Derived>& operator=(
       RxSO3Base<OtherDerived> const& other) {
-    quaternion() = other.quaternion();
+    quaternion_nonconst() = other.quaternion();
     return *this;
   }
```

**Comment by strasdat** [comment]
I thought I had unit test coverage for this but this bug is more subtle than it seems. In particular, this assignment-like operator is almost never called, since the default assignment operator was still generated by the compiler.

**Comment by strasdat** [comment]
Solved in PR #136.
